edge/cmd/edgemark/hollow_edgecore.go

```diff
@@ -0,0 +1,153 @@
+	"github.com/kubeedge/kubeedge/edge/pkg/metamanager"
+	"github.com/kubeedge/kubeedge/pkg/apis/componentconfig/edgecore/v1alpha1"
+	"github.com/kubeedge/kubeedge/pkg/version/verflag"
+)
+
+type hollowEdgeNodeConfig struct {
+	Token           string
+	NodeName        string
+	HTTPServer      string
+	WebsocketServer string
+	NodeLabels      map[string]string
+}
+
+func main() {
+	command := newHollowEdgeNodeCommand()
+	if err := command.Execute(); err != nil {
+		os.Exit(1)
+	}
+}
+
+// newHollowEdgeNodeCommand creates a *cobra.Command object with default parameters
+func newHollowEdgeNodeCommand() *cobra.Command {
+	s := &hollowEdgeNodeConfig{
+		NodeLabels: make(map[string]string),
+	}
+
+	cmd := &cobra.Command{
+		Use:  "edgemark",
+		Long: "edgemark",
+		Run: func(cmd *cobra.Command, args []string) {
+			verflag.PrintAndExitIfRequested()
+			run(s)
+		},
+		Args: func(cmd *cobra.Command, args []string) error {
+			for _, arg := range args {
+				if len(arg) > 0 {
+					return fmt.Errorf("%q does not take any arguments, got %q", cmd.CommandPath(), args)
+				}
+			}
+			return nil
+		},
+	}
+
+	fs := cmd.Flags()
+	fs.AddGoFlagSet(flag.CommandLine) // for flags like --docker-only
+	s.addFlags(fs)
+
+	return cmd
+}
+
+func run(config *hollowEdgeNodeConfig) {
+	c := EdgeCoreConfig(config)
+
+	// use fake runtime service
+	edged.DefaultGetRuntimeService = GetFakeRuntimeAndImageServices
+
+	edged.Register(c.Modules.Edged)
+	edgehub.Register(c.Modules.EdgeHub, c.Modules.Edged.HostnameOverride)
+	metamanager.Register(c.Modules.MetaManager)
+
+	dbm.InitDBConfig(c.DataBase.DriverName, c.DataBase.AliasName, c.DataBase.DataSource)
+
+	// start all modules
+	core.Run()
+}
+
+func (c *hollowEdgeNodeConfig) addFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&c.Token, "token", "", "Token indicates the priority of joining the cluster for the edge.")
+	fs.StringVar(&c.NodeName, "name", "fake-node", "Name of this Hollow Node.")
+	fs.StringVar(&c.WebsocketServer, "websocket-server", "", "Server indicates websocket server address.")
+	fs.StringVar(&c.HTTPServer, "http-server", "", "HTTPServer indicates the server for edge to apply for the certificate.")
+	bindableNodeLabels := cliflag.ConfigurationMap(c.NodeLabels)
+	fs.Var(&bindableNodeLabels, "node-labels", "Additional node labels")
+}
+
+func EdgeCoreConfig(config *hollowEdgeNodeConfig) *v1alpha1.EdgeCoreConfig {
+	edgeCoreConfig := v1alpha1.NewDefaultEdgeCoreConfig()
+
+	// overWrite config
+	edgeCoreConfig.DataBase.DataSource = "/edgecore.db"
+	edgeCoreConfig.Modules.EdgeHub.Token = config.Token
+	edgeCoreConfig.Modules.EdgeHub.HTTPServer = config.HTTPServer
+	edgeCoreConfig.Modules.EdgeHub.WebSocket.Server = config.WebsocketServer
+
+	// use remote runtime for test, not run Dockershim
+	edgeCoreConfig.Modules.Edged.RuntimeType = "remote"
+	edgeCoreConfig.Modules.Edged.RemoteRuntimeEndpoint = "/run/containerd/containerd.sock"
+	edgeCoreConfig.Modules.Edged.RemoteImageEndpoint = "/run/containerd/containerd.sock"
```

Comment:
`GetFakeRuntimeAndImageServices` This method uses fake runtime, so there is no need to consider the runtime endpoint here?
And I suggest that the `RuntimeType` can be changed to `fake`. Will it lead to new problems?

Reply:
thanks for your review, change RuntimeType to fake and RemoteRuntimeEndpoint  to `/run/fake/fake.sock`,  becase the default value for RemoteRuntimeEndpoint  and RemoteImageEndpoint  is `unix:///var/run/dockershim.sock`, if we do not update it, 
the function `startDockerServer` will be called and edged exit due to error

https://github.com/kubeedge/kubeedge/blob/f6b6118b0f802d8a359187fab54137d1c2dc8349/edge/pkg/edged/edged.go#L653-L656

Reply:
okey